src/back/spv/writer.rs

```diff
@@ -744,12 +744,12 @@ impl Writer {
             crate::TypeInner::Scalar { kind, width } => {
                 instruction = match kind {
                     crate::ScalarKind::Sint => {
-                        self.instruction_type_int(id, width, Signedness::Signed)
+                        self.instruction_type_int(id, width << 3, Signedness::Signed)
```

Comment:
I think these functions need to be changed. They accept `u8` for width, which is technically neither what SPIR-V needs (`u32`) or what IR has (`Bytes`). We need to either make them receive `Bytes` and do the conversion internally (inside the functions), or switch the arguments to `spirv::Word`